src/legacy/core_plugins/data/public/expressions/lib/execute.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { fromExpression } from '@kbn/interpreter/target/common';
+import { DataAdapter, RequestAdapter } from 'ui/inspector/adapters';
+import { Adapters } from 'ui/inspector';
+import { getInterpreter } from '../services';
+import { Ast, IExpressionLoaderParams, IInterpreterResult } from './_types';
+
+/**
+ * The search context describes a specific context (filters, time range and query)
+ * that will be applied to the expression for execution. Not every expression will
+ * be effected by that. You have to use special functions
+ * that will pick up this search context and forward it to following functions that
+ * understand it.
+ */
+
+export class ExpressionDataHandler {
+  private abortController: AbortController;
+  private expression: string;
+  private ast: Ast;
+
+  private inspectorAdapters: Adapters;
+  private promise: Promise<IInterpreterResult>;
+
+  constructor(expression: string | Ast, params: IExpressionLoaderParams) {
+    if (typeof expression === 'string') {
+      this.expression = expression;
+      this.ast = fromExpression(expression);
+    } else {
+      this.ast = expression;
+      this.expression = '';
+    }
+
+    this.abortController = new AbortController();
+    this.inspectorAdapters = this.getActiveInspectorAdapters();
+
+    const getInitialContext = () => ({
```

Comment:
The only breaking change for the expression renderer react component is that it won't be possible to pass in a custom `getInitialContext` function anymore. Maybe I'm missing something, but is that limitation necessary?

The alternative would be adding `getInitialContext` to `IExpressionLoaderParams` instead of `searchContext`. Then there could also be initial contexts other than `kibana_context`. Not important for the usage in Lens though.

Reply:
you can pass in the SearchContext (which will be returned by getInitialContext).
for now lets keep it like this, if we find a usecase for sending in something else we can extend this in the future.